dexlib2/src/main/java/org/jf/dexlib2/writer/DexWriter.java

```diff
@@ -166,6 +170,22 @@ protected DexWriter(Opcodes opcodes) {
         this.annotationSection = sectionProvider.getAnnotationSection();
         this.annotationSetSection = sectionProvider.getAnnotationSetSection();
         this.encodedArraySection = sectionProvider.getEncodedArraySection();
+
+        overflowableSections = new IndexSection<?>[] {
+                //stringSection,            // supports jumbo indexes
+                typeSection,
+                //protoSection,             // reduntant check: cannot be larger than methodSection
```

Comment:
"cannot be larger than methodSection" Challenge accepted :D

Seriously though, while likely true in practice, it is possible for the proto section to be larger. See, e.g. the const-method-type instruction, which could reference proto items that are not included as part of a method descriptor.

Reply:
so this a bug and protoSection should be included in the set, ok, will fix.

i had to guess why you hadn't included that in your existing check, and i tried documenting the guess as the rationale for exclusion was not immediately obvious. but a good thing came out of it anyway, a squashed bug :)

Reply:
Yeah, in prior dex versions, it wouldn't have been independently overflowable. But now that proto ids are more of a separate thing, it is.

Reply:
protoSection now included in overflow check